katsdpsigproc/fill.py

```diff
@@ -43,20 +47,21 @@ def __init__(self, context, dtype, ctype, tuning=None):
 
     @classmethod
     @tune.autotuner(test={'wgs': 128})
-    def autotune(cls, context, dtype, ctype):
+    def autotune(cls, context: AbstractContext, dtype: np.dtype, ctype: str) -> Mapping[str, Any]:
         queue = context.create_tuning_command_queue()
         shape = (1048576,)
         data = accel.DeviceArray(context, shape, dtype=dtype)
 
-        def generate(wgs):
+        def generate(wgs: int) -> Callable[[int], float]:
             fn = cls(context, dtype, ctype, {'wgs': wgs}).instantiate(queue, shape)
             fn.bind(data=data)
             return tune.make_measure(queue, fn)
 
         return tune.autotune(generate, wgs=[64, 128, 256, 512])
 
-    def instantiate(self, *args, **kwargs):
-        return Fill(self, *args, **kwargs)
+    def instantiate(self, command_queue: AbstractCommandQueue, shape: Tuple[int, ...],
+                    allocator: Optional[accel.AbstractAllocator] = None) -> 'Fill':
```

Comment:
Does `'Fill'` get linked to the `Fill` class by mypy?

Reply:
Yes.